### ISSUE TITLE
OTA: End device (Requestor) crashes if it cannot reach the Provider

### ISSUE THREAD

**Issue by mykrupp** [project-chip/connectedhomeip#14321]
#### Problem

Expected behavior: announce-ota-provider should error out when no Provider node ID has been registered. One reason that the Provider node ID would not be registered is if the provider has not been commissioned over thread

When the end device receives the Announce OTA Provider command and the Provider node ID indicated in that command is not reachable, the end device crashes. The function HandleNodeIdResolve in Discovery_ImplPlatform.cpp is passed a DnssdService pointer that contains incomplete information. Attributes mKey, mData contain null fields and mDataSize seems uninitialized.

Steps to reproduce: 
Follow - [https://github.com/selissia/connectedhomeip/edit/master/examples/ota-requestor-app/efr32/README.md](url)
But do not commission provider

Call stack in Ozone: 
<img width="456" alt="Screen Shot 2022-01-24 at 1 42 30 PM" src="https://user-images.githubusercontent.com/95718139/151192506-9a664b60-46e9-4ceb-88b2-3920731c140f.png">

Hardfault register details:
<img width="1379" alt="Screen Shot 2022-01-24 at 12 47 43 PM" src="https://user-images.githubusercontent.com/95718139/151192537-d201a9ac-304f-4b4b-98e0-60dafdc1f425.png">

 #### Proposed Solution

Either a null check around the provider node ID fields, or an error code indicating that the provider has not been comissioned.

**Comment by holbrookt** [comment]
@selissia should this issue be assigned to you as it's efr32? If not, can you help assign?

**Comment by selissia** [comment]
Yes, this is a DNS issue not specific to EFR32. As such I initially assigned it to @vivien-apple but @woody-apple unassigned it, I'm not sure why.

**Comment by holbrookt** [comment]
Maybe an automated script? Feel free to reassign if you think that's right
